### PR TITLE
refactor(w3c/groups): improve docs + typescript

### PR DESCRIPTION
**Pull request by marcoscaceres**
Improved typescript and docs

### CODE REVIEW

File: src/w3c/group.js [hunk 1]
```diff
@@ -44,9 +52,10 @@ async function getMultipleGroupDetails(groups) {
 }
 
 /**
- * @param {string} group
- * @typedef {{ wgId: number, wg: string, wgURI: string, wgPatentURI: string, wgPatentPolicy: string, groupType: W3CGroupType }} GroupDetails
- * @returns {Promise<GroupDetails|undefined>}
+ * Fetches configuration details for a single group.
+ * @param {string} group A group identifier.
+ * @typedef {{ wgId: number, wg: string, wgURI: string, wgPatentURI: string, wgPatentPolicy: string, groupType: string }} GroupDetails
```
Comment (sidvishnoi) [inline src/w3c/group.js:57]: Unnecessary typedef here

File: src/w3c/group.js [hunk 2]
```diff
@@ -1,16 +1,20 @@
 // @ts-check
 /**
  * @module w3c/group
- *
- * `group` is a shorthand configuration option for specifying `wg`, `wgId`,
- * `wgURI`, and `wgPatentURI` options.
+ * The purpose of this module is to fetch and set the working group configuration details
```
Comment (sidvishnoi) [inline src/w3c/group.js:4]: Is this AI written?

Comment (marcoscaceres) [reply src/w3c/group.js:4]: Nope. Do I sound like a robot? 🤣

File: src/w3c/group.js (hunk 1 above)
Comment (marcoscaceres) [inline src/w3c/group.js:57]:
```suggestion
```

File: src/w3c/group.js (hunk 2 above)
Comment (marcoscaceres) [reply src/w3c/group.js:4]: This might have been co-pilot maybe.

File: src/w3c/group.js [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
+ * for the document using the W3C's groups API.
```
Comment (marcoscaceres) [inline src/w3c/group.js:5]:
```suggestion
```

File: src/w3c/group.js [hunk 4]
```diff
@@ -1,16 +1,19 @@
 // @ts-check
 /**
  * @module w3c/group
- *
- * `group` is a shorthand configuration option for specifying `wg`, `wgId`,
- * `wgURI`, and `wgPatentURI` options.
+ * The purpose of this module is to fetch and set the working group configuration details
```
Comment (marcoscaceres) [inline src/w3c/group.js:4]:
```suggestion
 * The purpose of this module is to fetch and set the working group configuration details.
```

File: src/type-helper.d.ts [hunk 5]
```diff
@@ -146,8 +146,20 @@ interface Conf {
   };
   /** The title of the document */
   title?: string;
+
+  /** Group */
```
Comment (marcoscaceres) [inline src/type-helper.d.ts:150]:
```suggestion
  /** W3C Group - see https://respec.org/w3c/groups */
```

**Review by sidvishnoi — APPROVED** [review]

**Pull request merged**